activestorage/lib/active_storage/attached/changes/create_one_of_many.rb

```diff
@@ -2,9 +2,16 @@
 
 module ActiveStorage
   class Attached::Changes::CreateOneOfMany < Attached::Changes::CreateOne #:nodoc:
+    attr_reader :attachments
+
+    def initialize(name, record, attachable, attachments)
+      super(name, record, attachable)
+      @attachments = attachments
+    end
+
     private
       def find_attachment
-        record.public_send("#{name}_attachments").detect { |attachment| attachment.blob_id == blob.id }
+        attachments.detect { |attachment| attachment.blob_id == blob.id }
```

Comment:
I have almost no experience with active storage source code but I wonder if it would make sense to query attachments here, like:
```ruby
ActiveStorage::Attachment.find_by(record: record, name: name, blob_id: blob.id)
```
It seems like we have everything we need to perform the query and we don't even need to load all records to find just one

Reply:
That was the first solution we found that worked (it was actually done with an `exists` but same idea). The issue is that this is inside `CreateOneOfMany` which will traverse all attachments so that creates an N+1. We added a bit on it on the PRs description I hope the explanation is clear, but basically this method is called a bunch of times per `#attach` call

Reply:
Got it, thanks for clarifying
I wonder if it would make sense to memoize the whole `find_attachment` call to avoid iterating over `attachments` each time `#find_attachment ` is called?

Reply:
There shouldn't be that many attachments there so I don't think it's really a performance constraint. We were already iterating over it on memory before